keras/ops/image.py

```diff
@@ -501,3 +501,157 @@ def map_coordinates(
         fill_mode,
         fill_value,
     )
+
+
+class PadToBoundingBox(Operation):
+    def __init__(
+        self,
+        offset_height,
+        offset_width,
+        target_height,
+        target_width,
+        check_dims=True,
+    ):
+        super().__init__()
+        self.offset_height = offset_height
+        self.offset_width = offset_width
+        self.target_height = target_height
+        self.target_width = target_width
+        self.check_dims = check_dims
+
+    def call(self, image):
+        return _pad_to_bounding_box(
+            image,
+            self.offset_height,
+            self.offset_width,
+            self.target_height,
+            self.target_width,
+            self.check_dims,
+        )
+
+    def compute_output_spec(self, image):
+        out_shape = (
+            image.shape[0],
+            self.target_height,
+            self.target_width,
+            image.shape[-1],
+        )
+        if len(image.shape) == 3:
+            out_shape = out_shape[1:]
+        return KerasTensor(
+            shape=out_shape,
+            dtype=image.dtype,
+        )
+
+
+def _pad_to_bounding_box(
+    image, offset_height, offset_width, target_height, target_width, check_dims
+):
+    image = backend.convert_to_tensor(image)
+    is_batch = True
+    image_shape = image.shape
+    if len(image_shape) == 3:
+        is_batch = False
+        image = backend.numpy.expand_dims(image, 0)
+    elif len(image_shape) != 4:
+        raise ValueError(
+            "'image' (shape %s) must have either 3 or 4 dimensions."
```

Comment:
Use f strings for formatting

Reply:
Sorry for the delayed response. I was caught up in a conference deadline. I have updated the code.